<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>842</width>
    <height>677</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Sparse Surface Adjustment 2D Viewer</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QWidget" name="widget_2" native="true">
      <property name="maximumSize">
       <size>
        <width>200</width>
        <height>16777215</height>
       </size>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QGroupBox" name="groupBox_2">
           <property name="title">
            <string>Visualization</string>
           </property>
           <widget class="QWidget" name="layoutWidget">
            <property name="geometry">
             <rect>
              <x>4</x>
              <y>26</y>
              <width>181</width>
              <height>200</height>
             </rect>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_4">
             <item>
              <layout class="QFormLayout" name="formLayout_2">
               <property name="fieldGrowthPolicy">
                <enum>QFormLayout::ExpandingFieldsGrow</enum>
               </property>
               <item row="0" column="1">
                <widget class="QDoubleSpinBox" name="doubleSpinBox_2">
                 <property name="minimum">
                  <double>0.010000000000000</double>
                 </property>
                 <property name="singleStep">
                  <double>0.010000000000000</double>
                 </property>
                </widget>
               </item>
               <item row="0" column="0">
                <widget class="QLabel" name="label_3">
                 <property name="minimumSize">
                  <size>
                   <width>100</width>
                   <height>0</height>
                  </size>
                 </property>
                 <property name="text">
                  <string>point size</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="label_7">
                 <property name="minimumSize">
                  <size>
                   <width>100</width>
                   <height>0</height>
                  </size>
                 </property>
                 <property name="text">
                  <string> level</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QSpinBox" name="spinBox_2"/>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_3">
               <item>
                <widget class="QCheckBox" name="checkBox">
                 <property name="text">
                  <string>vertices (C-P)</string>
                 </property>
                 <property name="shortcut">
                  <string>Ctrl+P</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QCheckBox" name="checkBox_2">
                 <property name="text">
                  <string>range data</string>
                 </property>
                 <property name="shortcut">
                  <string>Ctrl+L</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QCheckBox" name="checkBox_4">
                 <property name="text">
                  <string>normals</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QCheckBox" name="checkBox_5">
                 <property name="text">
                  <string>correspondences</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QCheckBox" name="checkBox_3">
                 <property name="enabled">
                  <bool>false</bool>
                 </property>
                 <property name="text">
                  <string>mesh</string>
                 </property>
                 <property name="shortcut">
                  <string>Ctrl+G</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox">
           <property name="title">
            <string>Optimization</string>
           </property>
           <widget class="QWidget" name="layoutWidget">
            <property name="geometry">
             <rect>
              <x>2</x>
              <y>30</y>
              <width>199</width>
              <height>281</height>
             </rect>
            </property>
            <layout class="QFormLayout" name="formLayout">
             <property name="fieldGrowthPolicy">
              <enum>QFormLayout::ExpandingFieldsGrow</enum>
             </property>
             <item row="0" column="0">
              <widget class="QLabel" name="label">
               <property name="minimumSize">
                <size>
                 <width>100</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>Iterations</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QSpinBox" name="spinBox">
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="value">
                <number>10</number>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="label_2">
               <property name="minimumSize">
                <size>
                 <width>100</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>Norm. Ext. Radius:</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QDoubleSpinBox" name="doubleSpinBox">
               <property name="decimals">
                <number>4</number>
               </property>
               <property name="minimum">
                <double>0.000100000000000</double>
               </property>
               <property name="maximum">
                <double>10.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.001000000000000</double>
               </property>
               <property name="value">
                <double>0.005000000000000</double>
               </property>
              </widget>
             </item>
             <item row="8" column="0">
              <widget class="QPushButton" name="pushButton">
               <property name="maximumSize">
                <size>
                 <width>100</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>optimize</string>
               </property>
               <property name="shortcut">
                <string>O</string>
               </property>
              </widget>
             </item>
             <item row="8" column="1">
              <widget class="QLabel" name="stateLabel">
               <property name="minimumSize">
                <size>
                 <width>80</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>ready</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
               <property name="margin">
                <number>4</number>
               </property>
              </widget>
             </item>
             <item row="3" column="0">
              <widget class="QLabel" name="label_4">
               <property name="minimumSize">
                <size>
                 <width>100</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>NS step size</string>
               </property>
              </widget>
             </item>
             <item row="3" column="1">
              <widget class="QDoubleSpinBox" name="doubleSpinBox_3">
               <property name="decimals">
                <number>4</number>
               </property>
               <property name="minimum">
                <double>0.000100000000000</double>
               </property>
               <property name="maximum">
                <double>10.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.001000000000000</double>
               </property>
               <property name="value">
                <double>0.002000000000000</double>
               </property>
              </widget>
             </item>
             <item row="5" column="0">
              <widget class="QLabel" name="label_5">
               <property name="minimumSize">
                <size>
                 <width>100</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>NS steps</string>
               </property>
              </widget>
             </item>
             <item row="5" column="1">
              <widget class="QDoubleSpinBox" name="doubleSpinBox_4">
               <property name="decimals">
                <number>0</number>
               </property>
               <property name="minimum">
                <double>0.000000000000000</double>
               </property>
               <property name="maximum">
                <double>100.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>1.000000000000000</double>
               </property>
               <property name="value">
                <double>5.000000000000000</double>
               </property>
              </widget>
             </item>
             <item row="7" column="0">
              <widget class="QLabel" name="label_6">
               <property name="minimumSize">
                <size>
                 <width>100</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>NS max angle diff</string>
               </property>
              </widget>
             </item>
             <item row="7" column="1">
              <widget class="QDoubleSpinBox" name="doubleSpinBox_5">
               <property name="decimals">
                <number>0</number>
               </property>
               <property name="minimum">
                <double>0.000000000000000</double>
               </property>
               <property name="maximum">
                <double>360.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>1.000000000000000</double>
               </property>
               <property name="value">
                <double>60.000000000000000</double>
               </property>
              </widget>
             </item>
             <item row="10" column="0">
              <widget class="QPushButton" name="pushButton_3">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="maximumSize">
                <size>
                 <width>100</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>Solve DA</string>
               </property>
               <property name="shortcut">
                <string>O</string>
               </property>
              </widget>
             </item>
             <item row="11" column="0">
              <widget class="QPushButton" name="pushButton_2">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="maximumSize">
                <size>
                 <width>100</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>Prune Vertices</string>
               </property>
               <property name="shortcut">
                <string>O</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="ssa::SSA2DglWidget" name="ssaGLWidget" native="true">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>16777215</height>
       </size>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>842</width>
     <height>27</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
    <addaction name="actionSave_FMap"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Alt+O</string>
   </property>
   <property name="softKeyRole">
    <enum>QAction::NoSoftKey</enum>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Alt+S</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Alt+Q</string>
   </property>
  </action>
  <action name="actionSave_FMap">
   <property name="text">
    <string>Save FMap</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ssa::SSA2DglWidget</class>
   <extends>QWidget</extends>
   <header>ssa/apps/ssa_viewer_2d/ssa_2d_qglviewer_widget.h</header>
   <container>1</container>
   <slots>
    <slot>setOptimize()</slot>
    <slot>setIterations(int)</slot>
    <slot>setDAMaxDist(double)</slot>
    <slot>setPointSize(double)</slot>
    <slot>openFile()</slot>
    <slot>setStrategie(int)</slot>
    <slot>setDrawVertices(int)</slot>
    <slot>setDrawLandmarkVertices(int)</slot>
    <slot>setDrawNormals(int)</slot>
    <slot>setDrawCorrespondences(int)</slot>
    <slot>setDrawMesh(int)</slot>
    <slot>setDrawGrid(int)</slot>
    <slot>saveFile()</slot>
    <slot>saveFMap()</slot>
    <slot>prune()</slot>
    <slot>solveDA()</slot>
    <slot>setDrawLevel(int)</slot>
    <slot>Gen3DObjectList_update()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>ssaGLWidget</receiver>
   <slot>setOptimize()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>65</x>
     <y>95</y>
    </hint>
    <hint type="destinationlabel">
     <x>497</x>
     <y>338</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>ssaGLWidget</receiver>
   <slot>setIterations(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>138</x>
     <y>39</y>
    </hint>
    <hint type="destinationlabel">
     <x>497</x>
     <y>338</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBox_2</sender>
   <signal>valueChanged(double)</signal>
   <receiver>ssaGLWidget</receiver>
   <slot>setPointSize(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>142</x>
     <y>59</y>
    </hint>
    <hint type="destinationlabel">
     <x>522</x>
     <y>338</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>activated()</signal>
   <receiver>ssaGLWidget</receiver>
   <slot>openFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>522</x>
     <y>338</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox</sender>
   <signal>stateChanged(int)</signal>
   <receiver>ssaGLWidget</receiver>
   <slot>setDrawVertices(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>96</x>
     <y>89</y>
    </hint>
    <hint type="destinationlabel">
     <x>522</x>
     <y>338</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_2</sender>
   <signal>stateChanged(int)</signal>
   <receiver>ssaGLWidget</receiver>
   <slot>setDrawLandmarkVertices(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>89</x>
     <y>117</y>
    </hint>
    <hint type="destinationlabel">
     <x>522</x>
     <y>338</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_3</sender>
   <signal>stateChanged(int)</signal>
   <receiver>ssaGLWidget</receiver>
   <slot>setDrawMesh(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>66</x>
     <y>151</y>
    </hint>
    <hint type="destinationlabel">
     <x>522</x>
     <y>338</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_4</sender>
   <signal>stateChanged(int)</signal>
   <receiver>ssaGLWidget</receiver>
   <slot>setDrawNormals(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>108</x>
     <y>151</y>
    </hint>
    <hint type="destinationlabel">
     <x>522</x>
     <y>338</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_5</sender>
   <signal>stateChanged(int)</signal>
   <receiver>ssaGLWidget</receiver>
   <slot>setDrawCorrespondences(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>101</x>
     <y>181</y>
    </hint>
    <hint type="destinationlabel">
     <x>522</x>
     <y>338</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>activated()</signal>
   <receiver>ssaGLWidget</receiver>
   <slot>saveFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>522</x>
     <y>338</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave_FMap</sender>
   <signal>activated()</signal>
   <receiver>ssaGLWidget</receiver>
   <slot>saveFMap()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>522</x>
     <y>338</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>ssaGLWidget</receiver>
   <slot>setDAMaxDist(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>142</x>
     <y>68</y>
    </hint>
    <hint type="destinationlabel">
     <x>522</x>
     <y>338</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>ssaGLWidget</receiver>
   <slot>solveDA()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>78</x>
     <y>562</y>
    </hint>
    <hint type="destinationlabel">
     <x>522</x>
     <y>338</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>ssaGLWidget</receiver>
   <slot>prune()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>64</x>
     <y>590</y>
    </hint>
    <hint type="destinationlabel">
     <x>522</x>
     <y>338</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_2</sender>
   <signal>valueChanged(int)</signal>
   <receiver>ssaGLWidget</receiver>
   <slot>setDrawLevel(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>144</x>
     <y>96</y>
    </hint>
    <hint type="destinationlabel">
     <x>522</x>
     <y>338</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_2</sender>
   <signal>editingFinished()</signal>
   <receiver>ssaGLWidget</receiver>
   <slot>Gen3DObjectList_update()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>144</x>
     <y>96</y>
    </hint>
    <hint type="destinationlabel">
     <x>522</x>
     <y>338</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
